File: app/src/main/java/com/crossbowffs/quotelock/utils/DpUtils.java
package com.crossbowffs.quotelock.utils;

import android.content.Context;
import android.content.res.Resources;
import android.util.TypedValue;

/**
 * @author Yubyf
 */
public class DpUtils {

    public static float dp2px(Context context, float dip) {
        Resources r = context.getResources();
        return TypedValue.applyDimension(
                TypedValue.COMPLEX_UNIT_DIP,
                dip,
                r.getDisplayMetrics()
        );
    }
}

File: app/src/main/java/com/crossbowffs/quotelock/utils/DpUtils.kt
@file:JvmName("DpUtils")

package com.crossbowffs.quotelock.utils

import android.content.res.Resources
import android.util.TypedValue
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

private val displayMetrics = Resources.getSystem().displayMetrics

fun Float.dp2px(): Float = TypedValue.applyDimension(
    TypedValue.COMPLEX_UNIT_DIP,
    this,
    displayMetrics
)

fun Int.dp2px(): Float = TypedValue.applyDimension(
    TypedValue.COMPLEX_UNIT_DIP,
    this.toFloat(),
    displayMetrics
)

fun Int.px2dp(): Dp = (this.toFloat() / displayMetrics.density).dp